pilot/pkg/networking/plugin/authz/authorization.go

```diff
@@ -58,6 +58,17 @@ func NewBuilderSkipIdentity(actionType ActionType, push *model.PushContext, prox
 	return &Builder{builder: b}
 }
 
+// NewBuilderWithOptions allows a builder with custom defined options
+func NewBuilderWithOptions(actionType ActionType, push *model.PushContext, proxy *model.Proxy, option builder.Option) *Builder {
+	tdBundle := trustdomain.NewBundle(push.Mesh.TrustDomain, push.Mesh.TrustDomainAliases)
+	policies := push.AuthzPolicies.ListAuthorizationPolicies(proxy.ConfigNamespace, proxy.Labels)
+	if !util.IsIstioVersionGE117(proxy.IstioVersion) {
```

Comment:
Can you pls add a comment on why this check is needed? I assume it is prior to 1.17, it can't support HBONE properly?

Reply:
NewBuilderWithOptions is pretty much the same as NewBuilderSkipIdentity, where that conditional exists. The conditional comes from: https://github.com/istio/istio/pull/42666 -- seems to affect using io.istio.peer_principle vs RBAC principle.